htsget-config/src/storage/s3.rs

```diff
@@ -6,15 +7,23 @@ pub struct S3Storage {
   pub(crate) bucket: String,
   pub(crate) endpoint: Option<String>,
   pub(crate) path_style: bool,
+  #[serde(flatten)]
+  pub(crate) object_type: ObjectType,
```

Comment:
🤔 I thought that the "ObjectType" concept/naming was gone in favor of more concrete semantics?

Reply:
`object_type` is gone from the perspective of the user, for example, see this config: https://github.com/umccr/htsget-rs/blob/a34e3326273110f88120609cd1b146384da32be6/htsget-config/examples/config-files/c4gh.toml#L14-L15

`ObjectType` is necessary in code here because I needed a struct to represent additional configuration options that was common between `LocalStorage`, `S3Storage` and `UrlStorage`. I didn't want to repeat the following in all those structs:

```rust
pub struct ... {
    private_key: ...
    recipient_public_key: ...
}
```

I also don't think placing `C4GHPath` directly inside the storage config structs is good either because then `#[cfg(feature="experimental")]` would be everywhere, hence the added indirection.

I'm happy to rename `ObjectType` to something else if you think so?

Reply:
Hm, I see, if all `ObjectType` does is provide a "supertype" for `*Storage` structs, perhaps `StorageType` is a more fitting name? Ah, naming is hard :/